source/plugins/ruby/KubernetesApiClient.rb

```diff
@@ -97,6 +99,7 @@ def getKubeResourceInfoV2(resource, api_group: nil)
             Net::HTTP.start(uri.host, uri.port, :use_ssl => true, :ca_file => @@CaFile, :verify_mode => OpenSSL::SSL::VERIFY_PEER, :open_timeout => 20, :read_timeout => 40) do |http|
               kubeApiRequest = Net::HTTP::Get.new(uri.request_uri)
               kubeApiRequest["Authorization"] = "Bearer " + getTokenStr
+              kubeApiRequest["User-Agent"] = @@userAgent
               @Log.info "KubernetesAPIClient::getKubeResourceInfoV2 : Making request to #{uri.request_uri} @ #{Time.now.utc.iso8601}"
               response = http.request(kubeApiRequest)
```

Comment:
we should add useragent for watch api calls also - https://github.com/microsoft/Docker-Provider/blob/ci_prod/source/plugins/ruby/KubernetesApiClient.rb#L944